utils/runtime/getting_failure_step.yml

```diff
@@ -0,0 +1,27 @@
+- block:
+
+    ## Getting failure step from experiment-pod
+    - name: Getting failure step from experiment pod
+      shell: >
+        kubectl logs {{ chaos_pod_name }} -n {{ a_ns }} | grep "FAILED!" -B 1 | head -1 | awk -F "Step:" '{print $2}'
+      register: fail_step
+      args: 
+        executable: /bin/bash
+
+    - name: Record the name of failed step
+      set_fact:
+        failStep: "{{ fail_step.stdout }}"
+
+    - name: Print the name of failed step
+      debug:
+        msg: "failStep: {{ failStep }}"
+
+  when: "flag is defined and flag == 'Fail'"
+
+- block: 
+  
+    - name: Record the name of failed step
+      set_fact:
+        failStep: "N/A"
+
+  when: "flag is defined and flag == 'Pass'"
```

Comment:
If it is a single task can we write when with the task itself?

Reply:
There will be upcoming tasks added into that block(events)